src/NodeHandler.ts

```diff
@@ -0,0 +1,51 @@
+import { Evented } from '@dojo/core/Evented';
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import Map from '@dojo/shim/Map';
+import { NodeHandlerInterface } from './interfaces';
+
+/**
+ * Enum to identify the type of event.
+ * Listening to 'Projector' will notify when projector is created or updated
+ * Listening to 'Widget' will notifiy when widget root is created or updated
+ */
+export enum NodeEventType {
+	Projector = 'Projector',
+	Widget = 'Widget'
+}
+
+export class NodeHandler extends Evented implements NodeHandlerInterface {
+
+	private _nodeMap = new Map<string, HTMLElement>();
+
+	public get(key: string): HTMLElement | undefined {
+		return this._nodeMap.get(key);
+	}
+
+	public has(key: string): boolean {
+		return this._nodeMap.has(key);
+	}
+
+	public add(element: HTMLElement, properties: VNodeProperties) {
```

Comment:
nit: can you add return type of `: void` please?

Reply:
In other places through the PR also?

Reply:
Maybe it's okay because it is defined on the interface, I don't know what the standards are around this. Can't hurt to add them I guess.